### PR TITLE
Moved to dataclass for easier handling and better readability

### PR DESCRIPTION
**Pull request by xvzf**
@JonasVautherin FYI

### CODE REVIEW

File: pb_plugins/dcsdkgen/methods.py [hunk 1]
```diff
@@ -35,29 +43,35 @@ def __init__(
...
             raise Exception(
-                "Responses cannot have more than 1 return parameter (and an optional '*Result')!\nError in {}".format(method_output))
+                "Responses cannot have more than 1 return parameter" +
+                f"(and an optional '*Result')!\nError in {method_output}")
 
         if len(return_params) == 1:
-            self._return_type = extract_string_type(return_params[0])
-            self._is_return_type_primitive = is_primitive_type(
-                return_params[0])
+            self._return_type = \
```
[the remark this answers is not in the document]
Comment (xvzf) [reply pb_plugins/dcsdkgen/methods.py]: At least it doesn't complain - the original looked a bit ugly, in my opinion this is a bit more readable!

File: pb_plugins/dcsdkgen/methods.py [hunk 2]
```diff
@@ -35,29 +43,35 @@ def __init__(
     def extract_params(self, pb_method, requests):
         method_input = pb_method.input_type.split(".")[-1]
         request = requests[method_input]
-        nb_fields = len(request.field)
+        # nb_fields = len(request.field)
```
[the remark this answers is not in the document]
Comment (xvzf) [reply pb_plugins/dcsdkgen/methods.py]: It was hinted as an unused variable, so... I did a testrun and it worked fine! I thought you want to use it later so I just commented it out!

File: pb_plugins/dcsdkgen/methods.py [hunk 3]
```diff
@@ -35,29 +43,35 @@ def __init__(
...
+                    name=field.name,
+                    type=extract_string_type(field),
+                    is_primitive=is_primitive_type(field))
+                    )
 
     def extract_return_type_and_name(self, pb_method, responses):
         method_output = pb_method.output_type.split(".")[-1]
         response = responses[method_output]
-        nb_fields = len(response.field)
+        # nb_fields = len(response.field)
```
[the remark this answers is not in the document]
Comment (xvzf) [reply pb_plugins/dcsdkgen/methods.py]: Same as above!

File: pb_plugins/dcsdkgen/methods.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JonasVautherin) [reply pb_plugins/dcsdkgen/methods.py]: I am usually against comments :D. I just forgot to remove it ;-).

[the remark this answers is not in the document]
Comment (xvzf) [reply pb_plugins/dcsdkgen/methods.py]: Removed it!

**Pull request merged**